main.go

```diff
@@ -100,30 +101,65 @@ func runUp(nodeType string) {
 	var cmd string
 	baseCmd := fmt.Sprintf(`docker run --privileged -d -v /tmp/dind:/var/lib/docker %s -p %d:80 -p %d:443`, volumeFlag, port, tlsPort)
 
-	if nodeType == "creator-node" {
+	if nodeType == "discovery-provider" {
+		// server nginx runs on port 5000 and we should expose that
+		baseCmd = baseCmd + " -p 5000:5000"
+	}
+
+	switch nodeType {
+	case "creator-node":
 		cmd = fmt.Sprintf(baseCmd + ` \
         --name creator-node \
         -v /var/k8s/mediorum:/var/k8s/mediorum \
         -v /var/k8s/creator-node-backend:/var/k8s/creator-node-backend \
         -v /var/k8s/creator-node-db:/var/k8s/creator-node-db \
         audius/audius-docker-compose:` + imageTag)
-	} else {
+	case "discovery-provider":
 		cmd = fmt.Sprintf(baseCmd + ` \
         --name discovery-provider \
         -v /var/k8s/discovery-provider-db:/var/k8s/discovery-provider-db \
         -v /var/k8s/discovery-provider-chain:/var/k8s/discovery-provider-chain \
         audius/audius-docker-compose:` + imageTag)
+	case "identity-service":
+		cmd = fmt.Sprintf(baseCmd + ` \
+        --name identity-service \
+        audius/audius-docker-compose:` + imageTag)
+	default:
+		exitWithError(fmt.Sprintf("provided node type is not supported: %s", nodeType))
 	}
 
-	execCmd := fmt.Sprintf(`docker exec %s sh -c "while ! docker ps &> /dev/null; do echo 'starting up' && sleep 1; done && cd %s && docker compose up -d"`, nodeType, nodeType)
-
-	if err := runCommand("/bin/sh", "-c", cmd+" && "+execCmd); err != nil {
+	if err := runCommand("/bin/sh", "-c", cmd); err != nil {
 		exitWithError("Error executing command:", err)
 	}
+
+	awaitDockerStart()
+	audiusCli("set-network", network)
+
+	if nodeType == "discovery-provider" {
+		configureChainSpec()
+	}
+
+	switch nodeType {
+	case "creator-node":
+		execCmd := fmt.Sprintf(`docker exec %s sh -c "cd %s && docker compose up -d"`, nodeType, nodeType)
+		if err := runCommand("/bin/sh", "-c", execCmd); err != nil {
+			exitWithError("Error executing command:", err)
+		}
+	case "discovery-provider":
+		audiusCli("launch-chain")
```

Comment:
chain node can run w/o discovery, launch it first so it's present before the rest of the stack